### DISCUSSION TITLE
Should `generic_hygrostat` expose a device instead of an entity?

### DISCUSSION THREAD

**Discussion by KidEinstein** [home-assistant/architecture#783] [Entity Models]
Should `generic_hygrostat` expose a device instead of just an entity? The current humidity would then be exposed as a sensor within the device. This would allow integrations like HomeKit to pick up the current humidity correctly using [this](https://github.com/home-assistant/core/blob/e9697872c8ebb8e0b41c5e7bf78dab6d3b58afcf/homeassistant/components/homekit/__init__.py#L954-L961) code.

This discussion stemmed from a review comment on my pull request [here](https://github.com/home-assistant/core/pull/75700#discussion_r928329242). @MartinHjelmare mentioned that this change would require a config flow to set up the integration.

**Comment by Shulyaka** [comment]
I would say it's a broader question: how can virtual devices use device registry?

**Reply by KidEinstein** [reply]
@Shulyaka , did you want to suggest what would be a good next step? Is there anyone who could help decide whether virtual devices can use the device registry?

**Reply by balloob** [reply]
Virtual devices can use device registry by using their config entry ID as a unique ID.

**Reply by KidEinstein** [reply]
Thanks, @balloob . I think @Shulyaka had some concerns with using the device registry for virtual devices, [here](https://github.com/home-assistant/frontend/pull/14645#discussion_r1046018227). Copying the concerns here:

> 1. Register generic_hygrostat into device registry. As of now, however, the device registry is for real devices, not virtual ones.
> 2. Add the ability for generic_hygrostat to either register the current humidity sensor under its device in the device registry (can an entity belong to multiple devices? If not yet, that's another obstacle), or create a copy sensor (which is probably not the best workaround either)

From what you said, it looks like we should OK using the device registry for virtual devices.

I can think the second concern does need more discussion. Should we allow a sensor entity to be part of multiple virtual devices?
